src/cargo/ops/cargo_add/mod.rs

```diff
@@ -537,19 +538,67 @@ fn get_latest_dependency(
                     std::task::Poll::Pending => registry.block_until_ready()?,
                 }
             };
-            let latest = possibilities
-                .iter()
-                .max_by_key(|s| {
-                    // Fallback to a pre-release if no official release is available by sorting them as
-                    // less.
+
+            fn find_latest<'a>(
+                possibilities: impl Iterator<Item = &'a Summary>,
+            ) -> Option<&'a Summary> {
+                possibilities.max_by_key(|s| {
+                    // Fallback to a pre-release if no official release is available by sorting them
+                    // as less.
                     let stable = s.version().pre.is_empty();
                     (stable, s.version())
                 })
-                .ok_or_else(|| {
-                    anyhow::format_err!(
-                        "the crate `{dependency}` could not be found in registry index."
-                    )
-                })?;
+            }
+
+            let mut latest = find_latest(possibilities.iter()).ok_or_else(|| {
+                anyhow::format_err!(
+                    "the crate `{dependency}` could not be found in registry index."
+                )
+            })?;
+
+            if config.cli_unstable().msrv_policy {
```

Comment:
We should also have an unstable ` --ignore-rust-version` for bypassing this.

Reply:
That seems fine. Would that require some sort of formal process (e.g. RFC, separate PR, tracking issue, etc.)? I can also see where it might be nice to have a general flag that overrides MSRV behavior for all relevant operations, so this might be bigger than just cargo add.

Reply:
`--ignore-rust-version` already exists for `cargo check` and related commands which is what gave me the idea for suggesting this.

What I forgot to bring up and likely doesn't have to happen in this PR is a `--rust-version` flag that can override `package.rust-version`.  I would personally chalk it all up under this experimental feature of `msrv-policy` which are are using to collect feedback to prepare for an RFC.